consensus/src/simplex/actors/voter/actor.rs

```diff
@@ -159,8 +159,16 @@ impl<E: Clock, S: Scheme, D: Digest> Round<E, S, D> {
     }
 
     pub fn set_leader(&mut self, seed: Option<S::Seed>) {
-        let leader = select_leader::<S, _>(self.scheme.participants().as_ref(), self.round, seed);
-        self.leader = Some(leader);
+        let leader_idx =
+            select_leader::<S, _>(self.scheme.participants().as_ref(), self.round, seed);
```

Comment:
In the other place this function is used, it also needs the pub key. Perhaps just have this function return `(usize, &P)` so that you don't have to do `let leader = self.scheme.participants().key(...)...`

Reply:
good call, will change that now